frappe/utils/data.py

```diff
@@ -246,7 +246,8 @@ def format_datetime(datetime_string, format_string=None):
 
 def global_date_format(date):
 	"""returns date as 1 January 2012"""
-	formatted_date = getdate(date).strftime("%d %B %Y")
+	date = getdate(date)
+	formatted_date = babel.dates.format_date(date, locale=(frappe.local.lang or "").replace("-", "_"))
```

Comment:
Changes the format of the date to short version:
![image](https://cloud.githubusercontent.com/assets/9355208/26717960/e6ac4050-479c-11e7-837e-4894d2d0389b.png)

Maybe we can use `format='long'` parameter ?